### ISSUE TITLE
[RX 480] "Quality" Preset Results In Input Lag / Unsmooth Visual Gameplay

### ISSUE THREAD

**Issue by Xbye** [Xaymar/OBS-AMD-Advanced-Media-Framework#118]
This may be an issue centered around my RAM or motherboard though, as I only have 8GB RAM and my motherboard seems to be pretty terrible. I have a 4790k @ 4.5ghz and an RX 480 8GB OC from XFX.

**Specs**:
[Discord Image](https://cdn.discordapp.com/attachments/236980395386732554/241970609905467392/unknown.png)

**OBS Version**:
OBS 0.16.2 (64bit, windows)

**Plugin Version**:
09:40:38.762: [AMF Encoder] Version 1.3.3.1-1.3.0.5

**AMD Driver**:
non-whql-win10-64bit-radeon-software-crimson-16.10.1-oct6

**LOG / Video**:
https://www.twitch.tv/3ybx/v/97823861
https://gist.github.com/anonymous/04e0900a41fbcdc621fbde52a27e8754

(OLD)

> **x264**: Anything slower than Ultrafast (IE. Superfast/Veryfast/Faster...) -
> Influences framerate, causes input lag and my gameplay is visually unsmooth making tracking fast motion targets difficult.
> 
> **h264 Quality**:
> Does not influence framerate, but causes input lag and my gameplay is visually unsmooth making tracking fast motion targets difficult.
> 
> **h264 Balanced**:
> Doesn't even feel like I am streaming. Zero impact on input lag or visual gameplay.
> 
> I understand that "feeling" is not suffice enough evidence, but I do not own a high-speed camera to be able to test this. I've played fast motion (not CS:GO) FPS games for years and the arrival of h264 was a life saver for me when it didn't impact my gameplay at all. My resolution or framerate doesn't matter, only the preset matters which if it's not on balanced it will be too impactful.

**Comment by Xaymar** [comment]
Run OBS Studio as administrator, it should help with this issue. The encoder has no impact on input/gameplay unless the game itself uses hardware decoder/encoder features - which it often shouldn't.

**Comment by Xbye** [comment]
Is there any way for me to test if **Left4Dead 2** is using hardware decoder/encoder features?

**Comment by Xaymar** [comment]
Left 4 Dead 2 uses hardware decoding for the movies in the menu and intro. It might use some other decoding features for animated textures, not sure how much is actually being used there.

**Comment by Xbye** [comment]
Any fast-motion game you can recommend for me to test on then? See if the issue is related to just Left4Dead 2 or it's really on the encoder or not.

**Comment by Xaymar** [comment]
Overwatch, CS:GO, Beat Hazard.

**Comment by Xbye** [comment]
CS:GO is **not** fast motion at all, but it's the only game of the two that I own so I'll have to test it a bit then I guess.

**Comment by Xaymar** [comment]
You need to stop camping if CS:GO isn't fast motion at all or try some more fast paced game modes.

**Comment by kytos22** [comment]
@Xbye Try some aim map from the CSGO workshop, there is one where the bots can come from anywhere to knife you and you have to kill them. https://steamcommunity.com/sharedfiles/filedetails/?l=spanish&id=368026786 If this is not fast motion im russian, and i can tell you i'm not.

**Comment by Xbye** [comment]
@M4RK22 I've used that map. Still don't view CS:GO as a fast motion game.

I am still trying to solve the issue in CS:GO where my mouse movement feels floaty and not precise. Still can't fix it, so I am currently at an impass on being able to test CS:GO because I can't aim properly in that game.

**Comment by max20091** [comment]
Warframe is the best game to test recording performance since it don't have pre-rendered cutscene and have complex areas without eating much performance.
Try to use real fullscreen instead of borderless fullscreen will help improve performance

**Comment by kytos22** [comment]
@Xbye Enable direct input and disable mouse acceleration then adjust the mouse sensivity. Also don't limit the fps as this game gets weird input lag even at 60 fps.

**Comment by max20091** [comment]
@M4RK22 not all games support raw input and 60fps only give 16ms lag, not much for a normal game.
Also if use too much CPU, it will not have enough time to response, causing massive input lag!

**Comment by Xbye** [comment]
@M4RK22 @max20091 
I'm not an idiot. I know source engine well enough to mess with a bunch of stuff and nothing fixes it. I'll have to post on the CS:GO subreddit to see if there is something else.

I'll look into trying Warframe to test.

**Comment by locodicegr** [comment]
@Xbye alt+tab and go back in the game.

That fixes my weird mouse ''problem''

**Comment by kytos22** [comment]
@xbye i don't think you are an idiot, we try to help.  I've played cs:go for years, recorded, streamed with vce-amf (xaymar plugin + 480 + quality preset) the last week and i didn't have any input lag problem. Check your system resources and fps while streaming to see if there is some bottleneck there.

**Comment by Xbye** [comment]
@M4RK22 
There is no visible bottleneck. My CPU is well under 70%, GPU using whatever available resources it can, +300 FPS. I posted about this issue here as a bit of a gag joke:
http://steamcommunity.com/app/730/discussions/0/312265473878569842/
So you can see the amount of steps I've taken to try and solve this "issue".

@locodicegr 
I will try that.

**Comment by kytos22** [comment]
@Xbye OK, answered there.

When you say x264 you mean CPU, right? h264 AMF then? If so, report like the troubleshooting guide says https://github.com/Xaymar/OBS-AMD-Advanced-Media-Framework/wiki/Troubleshooting-Guide#finally-report-an-issue

**Comment by Xbye** [comment]
@Xaymar 
I have updated the OP. Whether or not you acknowledge it, I can feel it. Whether it's CS:GO or L4D2. It's not as easily noticeable when you are aiming at slow far-range targets (CS:GO) but when you get targets that are more consistently closer and you are constantly moving (L4D2) it's noticeable, even in CS:GO.

Furthermore, I stream at 60 FPS. There clearly were times during the stream where, even though the stats says 60 fps, the video was most definitely **not** 60 fps.

**Comment by Xaymar** [comment]
It's less about acknowledging it, it's more about it being reproducible. If it is, it's the encoder. If not, it's your setup/hardware.
As for "the video not being 60 FPS", if it says 60 FPS then it encoded at 60 FPS. OBS might have captured the same frame multiple times, you can limit game capture framerate if that happens.

**Comment by kytos22** [comment]
@Xbye For me that stream is fluid but it is posible you can notice some minimal stutter sometimes when streaming. I do when im streaming but 99% of the time I can play without problems or stutters with the quality setting 1080@60FPS - 9000Kbps CBR. The only thing different in my system is 16GB of RAM. It can be OBS at capture time, it takes a bit of performance. FYI there is a driver update 16.10.3, nothing to do with this but good to know.

**Comment by c2h5oh** [comment]
I'm trying to make my tests more more scientific - the recordings I linked in the ticket have each been done manually, they are off by a couple of frames, varying length and etc.

1. I'm going to record a couple video clips with lossless compression - I'm selecting a source right now: I want scenes with varying levels of motion, light, and complexity, each lasting at least 5-10s, all within a single, short video (around 1 min would be best..). 720p and 1080p at 30 and 60 fps. 1440p if it's possible to record 1440p without a 1440p screen by using AMD super resolution.

2. I'll use mpv with hardware decoding disabled and OpenGL or DirectX output to simulate game-like source

3. I'll capture the video with OBS with different settings that we can use for comparisons.

I own: 
* 7870 (VCE 1.0)
* 390x (VCE 2.0)
* Fury (VCE 3.0)

So we can do a comparison between 3 out of 4 supported right now. I might be able to borrow a R9 470 for full set, but I'm not sure right now - if any of you guys live in Toronto and would like to lend me one for one evening that would be great. Other hardware is beefy enough to take CPU and disk bandwidth out of the equation.

I'm missing 2 things:

1. Settings combinations that I should be testing (setting1: value1, value2; setting2: value3, value4) etc  @Xaymar this is something you're best suited to provide

2. A way to sync start/stop of video playback with recording start/stop with a single frame precision if possible. Is there something that would be more reliable than faking hotkeys? mpv supports lua scripting.

I need to be able to automate the whole thing. 2 (or 3) resolutions * 2 frame rates * 4-6 bitrates is already 16 to 36 runs per setting set, per card. Even with 1 minute video it's going to be hours - I need to be able to start it and go to work.

**Comment by Xaymar** [comment]
So turns out I can reproduce it. But only if I have B-Picture Reference active, use @jp9000 VRAM filling tool and play a game while doing so. If any of them are disabled or not running, it doesn't happen.
This is likely a problem on your end, with your card or with your driver. Try upgrading to 16.10.2 or 16.10.3.

**Comment by Shivansps** [comment]
@Xaymar Ive experienced what i think it is a similar issue
OBS 0.16.2 (64 bit Windows 10)
Plugin 1.3.3.1
Driver 16.10.3
Hardware:
Core I5-2500K
RX480 8GB

This is driving me crazy, the game is locked to 60 fps, and it can maintain it, this is the configuration im using:
http://i.imgur.com/h0x7JDF.png

I tried several things, im noticing 2 issues:
1) Bitrate gets kinda low at times, like <30, or even <20mbps, and i can notice it at several parts,  quality get better bitrate, still far from the target and with a no ideal quality.

2) Movement is BAD, REAL BAD, at times, ive found to way to fix it, ive uploaded a video
https://www.youtube.com/watch?v=oHfKVTAIUl4

Look at 0:25-0:35 just TERRIBLE, in-game is 100% fluid.

Locking both game and recording to 30 fps fixes it, for the most part, sometimes the movement will still look bad.
Dropping to balanced does not seem to help either.

**Comment by Xaymar** [comment]
@Shivansps Please open your own issue and follow the [Troubleshooting Guide](https://github.com/Xaymar/OBS-AMD-Advanced-Media-Framework/wiki/Troubleshooting-Guide).

Edit: Also only use "H264 Encoder (AMD Advanced Media Framework)", the other one is no longer supported and removed in a future update.

**Comment by Xaymar** [comment]
Bumping this issue since I changed the repository name.

**Comment by Xaymar** [comment]
I read the log file, however I do not see any signs of the encoder being overloaded. This likely means that your GPU is faulty or that your system is overloaded (too little bandwidth available). Best solution for you would be to not use the "Quality" Quality Preset and instead use "Balanced" or "Speed".

**Comment by Xbye** [comment]
> I read the log file, however I do not see any signs of the encoder being overloaded. This likely means that your GPU is faulty or that your system is overloaded (too little bandwidth available). Best solution for you would be to not use the "Quality" Quality Preset and instead use "Balanced" or "Speed".
> Great, faulty GPU/system... Sounds about right for my type of luck.

Do you know where the bandwidth would be limited at, like my RAM perhaps? My motherboard is pretty old I guess but idk how any of that encoding stuff works with my hardware.

**Comment by Xaymar** [comment]
It would most likely be the Southbridge since it handles most devices attached to the mainboard. Try reducing the amount of devices attached to the system through USB, SATA or similar.

**Comment by Xbye** [comment]
> It would most likely be the Southbridge since it handles most devices attached to the mainboard. Try reducing the amount of devices attached to the system through USB, SATA or similar.

Well, shit. This poses a problem then as every USB/SATA I have connected is essential besides my Webcam (which isn't active during the scene, but is "enabled" because the light turns on when I open OBS). Perhaps it's because I have it set to MJPEG? Probably isn't worth the quality since my webcam source is so small anyways.

I guess I'll have to test with my webcam unpluged maybe?

**Comment by Xaymar** [comment]
Webcam is probably the biggest bandwidth hog in a System. It's trial and error from here on out though :/

**Comment by Xbye** [comment]
Went over to the OBS IRC to try and get information on Motherboards/RAM being possible bottlenecks but was met with hostility. Outside of you, I can't find any information on what else can be a bottleneck besides my GPU/CPU, as this issue also applies to using x264 veryfast or superfast.

@Xaymar 
Here is 3 more logs, not sure if they're still useful or not;
https://gist.github.com/anonymous/cc33d6550ccc00bf199bfca26cf92242
https://gist.github.com/anonymous/64bfae0cdbbf3276099dad4cb6d80e94
https://gist.github.com/anonymous/c3179909aff2a6d5da42ed30d1710ff0

I used the two games I am the most familiar with movement and timing: Left4Dead 2 / CrossCode _(Monitor capture)_

I've disconnected my webcam didn't seem to the solve the issue. I lowered the polling rate of my keyboard from 1000hz to 500hz, still nothing. I then lowered it from 500hz to 250hz and that seemed to make the impact of the issue less noticeable, or non existent (could not tell which).

I have a Corsair Strafe. I then decided to plug in both USBs from my keyboard into the my motherboard and was able to use 500hz without issue in CrossCode and quality preset. However I am still unable to, even with 720@60 w/ Quality preset, in Left4Dead2 not feel the issue regardless of polling rate of my keyboard. In fact, the input delay is so noticeable it makes my mouse (1000hz polling rate) very floaty and definitely not usable competition wise.

This is my USB hub (Minus the webcam, removed it for testing):
https://cdn.discordapp.com/attachments/236980395386732554/243787682524823552/unknown.png
I think this issue honestly might be my older motherboard being the bottleneck, but I no one else with an rx 480 is really chipping in.

**Comment by Xaymar** [comment]
Same here, try 16.11.1 with 1.4.0.0 and report back.

**Comment by Xbye** [comment]
> Same here, try 16.11.1 with 1.4.0.0 and report back.

Sure, I'll get to that ASAP. By the way, do you think overlays (IE. Mumble/MSI Afterburner/RivaTuner) might also be possible culprits?

**Comment by Xaymar** [comment]
Depends on the overlay. There is  thread with something like an overlay on reddit: https://www.reddit.com/r/Amd/comments/5aj3v6/psa_if_you_are_having_bluescreens_crashing_or/

**Comment by kytos22** [comment]
Agree, some time ago the MSI gaming App (background service) was causing my gpu to reset to stock voltages all the time making my gpu unstable.

**Comment by Xbye** [comment]
@M4RK22 Idk if it might have been the culprit with my fan speeds a while back but I haven't had any issues, yet.

@Xaymar Updated to the newest plugin/drivers, but due to issues currently unable to further test this. Sorry ☹️

**Comment by Xbye** [comment]
**LOG**:
https://gist.github.com/anonymous/cc673ad7e3ced8b04ae7f718a50cca5c

Currently using 1.4.1.0. Issue does not seem to be resolved for me unfortunately. So going to try and disable my overlays and other hooking programs to test sometime soon.

_A reddit post:_

> MSI Afterburner can cause visual glitches in some games as well (particularly Dying light and Star Citizen) even if it is only installed and not running at the time.

If this is true, then I'm in trouble if MSI afterburner is the cause. I require RivaTuner to limit my FPS to use vertical sync without input lag when I have to stream tournament matches, and it's very useful for monitoring temperatures/fan speed/usage but I could forget about that part easily if need be.

**Comment by kytos22** [comment]
You can use the FRTC inside crimson to set the FPS target limit, unless it is limited for what you want to do. One thing is FPS limit and other Vsync. You can set your FPS to equal your monitor Hz but that does not mean that they are in sync. Unless afterburner have some magic that i don't know about.

**Comment by Xaymar** [comment]
Closing this now, as it isn't a plugin issue. If you need help with building or configuring a PC, there are many sites for that.

**Comment by Xbye** [comment]
@M4RK22 
I tried using FRTC inside crimson, and it is incredibly laggy as all hell. I don't know **what** it was doing, but it was making my game so input laggy I might as well have vsync on without frame-rate capping. I'll just stick to RivaTuner to cap my FPS for now.

@Xaymar 
I understand that you closed this issue, but I wanted to keep you involved in the progress I've made on this issue. I went through some testing and found out some of the issues. The culprits were impossible for me to detect in a game like CS:GO/Left4Dead2 since I get hundreds of FPS... but CrossCode you don't get much FPS so it was easy to notice when FPS changed because it's very sensitive.

**1.)** I was window capturing [NohBoard](https://obsproject.com/forum/resources/nohboard.44/), an app used to display keypresses on stream. Whenever this source was active in any scene, it caused CrossCode to drop about ~7 FPS. This was a leftover relic from when I upgraded from Win7, and am now using game capture for NohBoard.

**2.)** Though I solved the NohBoard issue, I was still losing ~7 FPS when moving to other scenes. I found out that having the **preview** feature of OBS was causing frame drops as well. Disabling OBS preview along with NohBoard change allowed me to stream both CrossCode and Left4Dead 2 without added input lag, however both of these are capped at 70 (refresh rate) FPS so I'll have to test uncapped FPS later with Left4Dead 2.

This was all done in **Version 1.3.3.1-1.3.0.5**, as the CBR is still broken in the current new version.
